drafts/draft-0005.md

```diff
@@ -0,0 +1,188 @@
+-----|------|------------
+`id` | string | Required. An identifier for the multisig request that is generated by the multisig service.
+`statusHref` | string | Required. A fully qualified url where an agent can poll for updates on the status of the multisig request. This is specified in the Status Endpoint section.
+`extras` | object | Optional. An object that contains additional application specific response data which could be used to provide an enhanced user experience. For instance, showing a link to authorize a transaction.
+
+Example:
+
+```json
+{
+  "id": "c585ceee-b1b9-4009-aa37-b8544346a036",
+  "statusHref": "https://stellarguard.me/api/transactions/c585ceee-b1b9-4009-aa37-b8544346a036",
+  "extras": {
+    "isStellarGuard": true,
+    "url": "https://stellarguard.me/transactions/c585ceee-b1b9-4009-aa37-b8544346a036"
+  }
+}
+```
+
+#### Status Endpoint
+
+This will be used to query for the status of a multisig transaction.
+
+##### Request
+
+`GET MULTISIG_ENDPOINT>/:id`
+
+Request Path Parameters:
+
+Name | Type | Description
+-----|------|------------
+`id` | string | Required. The id used to look up the status, which was returned in the submit transaction response.
+
+Example:
+
+```
+GET https://stellarguard.me/api/transactions/c585ceee-b1b9-4009-aa37-b8544346a036
+```
+
+##### Response
+
+On success, the endpoint should return a `200 OK` HTTP status code and a JSON-encoded object.
+
+Response fields:
+
+Name | Type | Description
+-----|------|------------
+`id` | string | Required. An identifier for the multisig request that is generated by the multisig service.
+`status` | string | Required. An string indicating the status of the multisig request which can have one of the following values:<br><br>- `pending` - The request requires more signatures<br>- `success` - The transaction has been fully signed<br>- `failed` - There was an error with the transaction and it could not be submitted successfully
+`uri` | string | Required. The SEP-0007 URI that was submitted to the service
+`signers`| string[] | Required. An array of public keys that have signed the transaction. 
+
+
+Example:
+
+```json
+{
+  "id": "c585ceee-b1b9-4009-aa37-b8544346a036",
+  "status": "pending",
+  "uri": "web+stellar:tx?xdr=AAAAAHFd0%2BHQV5u%2FY%2FfM3%2BVelUr1IWwSFL8CUDIAUudUdD4MAAAAZAADyI8AAAADAAAAAAAAAAAAAAABAAAAAAAAAAEAAAAAcV3T4dBXm79j98zf5V6VSvUhbBIUvwJQMgBS51R0PgwAAAAAAAAAAACYloAAAAAAAAAAAVR0PgwAAABAEw8ODG0iixkbAHg1aJATAnZS2531PhGauuSvFDad2WxHKzIenUNbc7K5mGiSpe5jvqe19OQCbNFuBjqN11jfBw%3D%3D",
+  "signers": ["GBYV3U7B2BLZXP3D67GN7ZK6SVFPKILMCIKL6ASQGIAFFZ2UOQ7AZB5V"]
+}
+```
+
+####  CORS headers
+
+In order to comply with browser cross-origin access policies, the service should provide wildcard CORS response HTTP header. The following HTTP header must be set for the API endpoints:
+
+```
+Access-Control-Allow-Origin: *
+```
+
+### Multisig Transaction Flow
+
+1. Determine whether a transaction requires additional signatures in order to be valid.
```

Comment:
Since the discussion on the list I think this should be removed. A point to solve: what must be the behavior if the transaction is valid? send it to coordinator anyway? send it to horizon? I think the coordinator must always be in charge of posting validated transactions for consistency (maybe it got its own way to deal with it).

Reply:
Actually 've said exactly the opposite: if an account did enable multisig coordinator, **then** it likely need it, **then** its transaction are unlikely to be valid at that point. **Even if** the transaction were valid, you would send it to coordinator in case it triggers side-effect (logging transaction) or in case in have a special treatment of them (sending to a particular endpoint, email notification, ...). Either you respect user configuration, either you don't. If you do, you clearly don't need to force wallet checking for transaction validity.

Reply:
The point is: a signature coordinators are going to act as a submissions service.

Reply:
If someone setup signature collection for its account he'll need it. I don't get how you can miss that.

My point was not about how people would use it, but about the fact that if you happen to send part of your transactions to multisig coordinator (for some reason it's not valid yet) and some other not (maybe you got the signature from somewhere else like preTxAuth) you're going to create inconsistencies into the coordinator service. Either you use it either you don't, but one time yes one time no looks like a good way to create a mess.

Now my proposal allow to make it consistent while to remove the bloat for wallet implementors to have to check for transaction validity, which is a non-trivial task really. You can keep this requirement if you want: but as I shown it's not really useful and it makes the SEP significantly harder to implement.

Reply:
I have not understood your response. Can you provide context?  — it’s confusing when you use pronouns like “it” then I do not know what you are referring to

By definition, a wallet is capable of checking if a transaction is valid, since it will submit to the network when the user hits submit. So what is the **additional cost** on a wallet implementer?

Reply:
I think most wallets are actually not able to check if a transaction is valid; I think most wallet doesn't even handle multi-signature at all. They just have 1-signers account, they sign and they send to Horizon knowing it is valid.

If you have the coordinator handle multisignature, maybe those wallet can join for free; If you don't they will not join because it's actually not so easy checking for transaction validity. And that is the additional cost I'm talking about. And for what? Being able to introduce a non-consistent behavior.

Reply:
Ok, we can update the first step in the workflow from `Determine whether a transaction requires additional signatures in order to be valid.` to `Determine whether the transaction has been signed by at least 1 signer`.

This will remove the requirement from wallets to perform checks on whether the transaction needs more signatures.

Reply:
That requirement is already listed as:
> Before submitting the transaction, it should contain at least one valid signature.

It becomes a bit awkward if I move it to the front because it means I need two "early exits". If you can think of a good way to word that I'll move it to the front, but it really doesn't which order it is in as long as all the steps are taken.

Reply:
Resolved by last update